<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.189">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>readme</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="README_files/libs/clipboard/clipboard.min.js"></script>
<script src="README_files/libs/quarto-html/quarto.js"></script>
<script src="README_files/libs/quarto-html/popper.min.js"></script>
<script src="README_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="README_files/libs/quarto-html/anchor.min.js"></script>
<link href="README_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="README_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="README_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="README_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="README_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">



<section id="somatic-shortv-nf" class="level1">
<h1>Somatic-shortV-nf</h1>
<p align="center">
:wrench: This pipeline is currently under development :wrench:
</p>
<ul>
<li><p><a href="#user-guide">User guide</a></p>
<ul>
<li><a href="#infrastructure-usage-and-recommendations">Infrastructure usage and recommendations</a></li>
</ul></li>
<li><p><a href="#benchmarking">Benchmarking</a></p></li>
<li><p><a href="#workflow-summaries">Workflow summaries</a></p>
<ul>
<li><a href="#metadata">Metadata</a></li>
<li><a href="#component-tools">Component tools</a></li>
</ul></li>
<li><p><a href="#additional-notes">Additional notes</a></p></li>
<li><p><a href="#helpfaqtroubleshooting">Help/FAQ/Troubleshooting</a></p></li>
<li><p><a href="#acknowledgementscitationscredits">Acknowledgements/citations/credits</a></p></li>
<li><p><a href="#description">Description</a></p></li>
</ul>
<p>Somatic-shortV-nf is a Nextflow pipeline for identifying somatic short variant events in Illumina short read whole genome sequence data. We have followed the <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035894731-Somatic-short-variant-discovery-SNVs-Indels-">GATK Best practices</a> . The input to this pipeline are the per sample “g.vcf” files which are generated using <a href="https://nf-co.re/sarek">nf-core/sarek</a>.</p>
<p>There are three main steps in the process of calling Somatic Short Variants:</p>
<ol type="1">
<li><strong>Creation of Somatic short variants Panel of Normals (PoN)</strong> : <br>This involves converting the Normal BAMs to PON. The PON’s are -</li>
</ol>
<ul>
<li>Made from normal samples i.e.&nbsp;the samples derived from healthy tissue and</li>
<li>Their main purpose is to capture recurrent technical artifacts in order to improve the results of the variant calling analysis.</li>
<li>Please use the github repository <a href="https://github.com/Sydney-Informatics-Hub/SomaticShortV_makePON-nf">SomaticShortV_makePON-nf</a> to create the PoN. <br></li>
</ul>
<ol start="2" type="1">
<li><strong>Call the somatic short variants for a Tumor-Normal pair</strong> : <br>The two main steps involved in calling Somatic Short Variants are -</li>
</ol>
<ul>
<li><p>Generate a large set of candidate somatic variants.</p></li>
<li><p>Filter the candidate somatic variants into a more confident set of somatic variant calls.</p></li>
<li><p>Calling candidate somatic short variants involved the use of the tool <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035531132">Mutect2</a> which calls both SNVs and indels simultaneously by generating a local assembly of haplotypes in an active region, de-novo. When Mutect2 sees a region with somatic variations, it dis-regards the existing mapping information completely and re-assembles the reads in that region in order to generate candidate variant haplotypes.</p></li>
<li><p>This was followed by calculating contamination using the tools <a href="https://gatk.broadinstitute.org/hc/en-us/articles/9570416554907-GetPileupSummaries">GetPileupSummaries</a> and <a href="https://gatk.broadinstitute.org/hc/en-us/articles/9570322332315-CalculateContamination">CalculateContamination</a></p></li>
<li><p>The next step was to learn the parameters of a model for orientation bias using the tool <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360051305331-LearnReadOrientationModel">LearnReadOrientationModel</a></p></li>
<li><p>The <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360036856831-FilterMutectCalls">FilterMutectCalls</a> tool then applies filters to the raw output of Mutect2.</p></li>
</ul>
<ol start="3" type="1">
<li><strong>Annotating the variants</strong> <br> An external genomic variant annotations and functional effect prediction tool <a href="http://pcingola.github.io/SnpEff/">SnpEff</a> was used for annotating the filtered variants (such as amino-acid changes etc). Please refer to the above link for SnpEff details.</li>
</ol>
<ul>
<li><a href="#diagram">Diagram</a></li>
</ul>
<p align="center">
<img src="./images/Somatic_variant_calling.png" width="60%">
</p>
<section id="workflow-description" class="level2">
<h2 class="anchored" data-anchor-id="workflow-description">Workflow description</h2>
</section>
<section id="user-guide" class="level2">
<h2 class="anchored" data-anchor-id="user-guide">User guide</h2>
<p>To run this pipeline, you will need to prepare your input files, reference data, and clone this repository. Before proceeding, ensure Nextflow is installed on the system you’re working on. To install Nextflow, see these <a href="https://www.nextflow.io/docs/latest/getstarted.html#installation">instructions</a>.</p>
<section id="prepare-inputs" class="level3">
<h3 class="anchored" data-anchor-id="prepare-inputs">1. Prepare inputs</h3>
<p>To run this pipeline you will need the following inputs:</p>
<ul>
<li>Paired <code>Tumor-Normal</code> BAM files</li>
<li>Corresponding BAM index files<br>
</li>
<li>Input sample sheet</li>
</ul>
<p>This pipeline processes paired BAM files and is capable of processing multiple samples in parallel. BAM files are expected to be coordinate sorted and indexed (see <a href="https://github.com/Sydney-Informatics-Hub/Fastq-to-BAM">Fastq-to-BAM</a> for an example of a best practice workflow that can generate these files).</p>
<p>You will need to create a sample sheet with information about the samples you are processing, before running the pipeline. This file must be <strong>tab-separated</strong> and contain a header and one row per sample. Columns should correspond to sampleID, BAM-N file, BAM-T file:</p>
<table class="table">
<thead>
<tr class="header">
<th>sampleID</th>
<th>bam-N</th>
<th>bam-T</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>SAMPLE1</td>
<td>/data/Bams/sample1-N.bam</td>
<td>/data/Bams/sample1-T.bam</td>
</tr>
<tr class="even">
<td>SAMPLE2</td>
<td>/data/Bams/sample2-N.bam</td>
<td>/data/Bams/sample2-T.bam</td>
</tr>
</tbody>
</table>
<p>When you run the pipeline, you will use the mandatory <code>--input</code> parameter to specify the location and name of the input file:</p>
<pre><code>--input /path/to/samples.tsv</code></pre>
</section>
<section id="prepare-the-reference-materials" class="level3">
<h3 class="anchored" data-anchor-id="prepare-the-reference-materials">2. Prepare the reference materials</h3>
<p>To run this pipeline you will need the following reference files:</p>
<ul>
<li>Indexed reference genome in FASTA format</li>
</ul>
<p>You will need to download and index a copy of the reference genome you would like to use. Reference FASTA files must be accompanied by a .fai index file. If you are working with a species that has a public reference genome, you can download FASTA files from the <a href="https://asia.ensembl.org/info/data/ftp/index.html">Ensembl</a>, <a href="https://genome.ucsc.edu/goldenPath/help/ftp.html">UCSC</a>, or <a href="https://www.ncbi.nlm.nih.gov/genome/doc/ftpfaq/">NCBI</a> ftp sites. You can use our <a href="https://github.com/Sydney-Informatics-Hub/IndexReferenceFasta-nf">IndexReferenceFasta-nf pipeline</a> to generate indexes.</p>
<p>When you run the pipeline, you will use the mandatory <code>--ref</code> parameter to specify the location and name of the reference.fasta file:</p>
<pre><code>--ref /path/to/reference.fasta</code></pre>
</section>
<section id="clone-this-repository" class="level3">
<h3 class="anchored" data-anchor-id="clone-this-repository">3. Clone this repository</h3>
<p>Download the code contained in this repository with:</p>
<pre><code>git clone https://github.com/Sydney-Informatics-Hub/Somatic-shortV-nf</code></pre>
<p>This will create a directory with the following structure:</p>
<pre><code>Somatic-shortV-nf/
├── LICENSE
├── README.md
├── config/
├── main.nf
├── modules/
└── nextflow.config</code></pre>
<p>The important features are:</p>
<ul>
<li><strong>main.nf</strong> contains the main nextflow script that calls all the processes in the workflow.</li>
<li><strong>nextflow.config</strong> contains default parameters to use in the pipeline.</li>
<li><strong>modules</strong> contains individual process files for each step in the workflow.</li>
<li><strong>config</strong> contains infrastructure-specific config files (this is currently under development)</li>
</ul>
</section>
<section id="run-the-pipeline" class="level3">
<h3 class="anchored" data-anchor-id="run-the-pipeline">4. Run the pipeline</h3>
<p>The most basic run command for this pipeline is:</p>
<pre><code>nextflow run main.nf --ref reference.fasta
</code></pre>
<p>By default, this will generate <code>work</code> directory, <code>results</code> output directory and a <code>runInfo</code> run metrics directory in the same location you ran the pipeline from.</p>
<p>To specify additional optional tool-specific parameters, see what flags are supported by running:</p>
<pre><code>nextflow run main.nf --help </code></pre>
<p>If for any reason your workflow fails, you are able to resume the workflow from the last successful process with <code>-resume</code>.</p>
</section>
<section id="results" class="level3">
<h3 class="anchored" data-anchor-id="results">5. Results</h3>
<p>Once the pipeline is complete, you will find all outputs for each sample in the results directory. Within each sample directory there is a subdirectory for each tool run which contains all intermediate files and results generated by each step.</p>
</section>
</section>
<section id="component-tools" class="level2">
<h2 class="anchored" data-anchor-id="component-tools">Component tools</h2>
<p>To run this pipeline you must have Nextflow and Singularity installed on your machine. All other tools are run using containers.</p>
<table class="table">
<thead>
<tr class="header">
<th>Tool</th>
<th style="text-align: left;">Version</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Nextflow</td>
<td style="text-align: left;">&gt;=20.07.1</td>
</tr>
<tr class="even">
<td>Singularity</td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td>SnpEff</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td>R</td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
</section>
<section id="additional-notes" class="level2">
<h2 class="anchored" data-anchor-id="additional-notes">Additional notes</h2>
</section>
<section id="help-faq-troubleshooting" class="level2">
<h2 class="anchored" data-anchor-id="help-faq-troubleshooting">Help / FAQ / Troubleshooting</h2>
<ul>
<li>It is essential that the reference genome you’re using contains the same chromosomes, contigs, and scaffolds as the BAM files. To confirm what contigs are included in your indexed BAM file, you can use Samtools idxstats:</li>
</ul>
<pre><code>samtools idxstats input.bam | cut -f 1</code></pre>
</section>
<section id="licenses" class="level2">
<h2 class="anchored" data-anchor-id="licenses">License(s)</h2>
</section>
<section id="acknowledgementscitationscredits" class="level2">
<h2 class="anchored" data-anchor-id="acknowledgementscitationscredits">Acknowledgements/citations/credits</h2>
<section id="authors" class="level3">
<h3 class="anchored" data-anchor-id="authors">Authors</h3>
<ul>
<li>Tracy Chew (Sydney Informatics Hub, University of Sydney)</li>
<li>Cali Willet (Sydney Informatics Hub, University of Sydney)</li>
<li>Nandan Deshpande (Sydney Informatics Hub, University of Sydney)</li>
</ul>
</section>
<section id="acknowledgements" class="level3">
<h3 class="anchored" data-anchor-id="acknowledgements">Acknowledgements</h3>
<ul>
<li>This pipeline was built using the <a href="https://github.com/Sydney-Informatics-Hub/Nextflow_DSL2_template">Nextflow DSL2 template</a>.<br>
</li>
<li>Documentation was created following the <a href="https://github.com/AustralianBioCommons/doc_guidelines">Australian BioCommons documentation guidelines</a>.</li>
</ul>
</section>
<section id="cite-us-to-support-us" class="level3">
<h3 class="anchored" data-anchor-id="cite-us-to-support-us">Cite us to support us!</h3>
<p>Acknowledgements (and co-authorship, where appropriate) are an important way for us to demonstrate the value we bring to your research. Your research outcomes are vital for ongoing funding of the Sydney Informatics Hub and national compute facilities. We suggest including the following acknowledgement in any publications that follow from this work:</p>
<p>The authors acknowledge the technical assistance provided by the Sydney Informatics Hub, a Core Research Facility of the University of Sydney and the Australian BioCommons which is enabled by NCRIS via Bioplatforms Australia.</p>
</section>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>